src/clp_ffi_py/modules/clp_four_byte_encoder.cpp

```diff
@@ -0,0 +1,102 @@
+        "message.\n"
+        ":param timestamp_format: Timestamp format to be use when generating the logs with a "
+        "reader.\n"
+        ":param timezone: Timezone in TZID format to be use when generating the timestamp "
+        "from Unix epoch time.\n"
+        ":raises NotImplementedError: If metadata length too large.\n"
+        ":return: The encoded preamble.\n");
+
+PyDoc_STRVAR(
+        encode_message_and_timestamp_delta_doc,
+        "encode_message_and_timestamp_delta(timestamp_delta, msg)\n"
+        "--\n\n"
+        "Encodes the log `msg` along with the timestamp delta.\n"
+        ":param timestamp_delta: Timestamp difference in miliseconds between the current log "
+        "message and the previous log message.\n"
+        ":param msg: Log message to encode.\n"
+        ":raises NotImplementedError: If the log message failed to encode, or the timestamp delta "
+        "exceeds the supported size.\n"
+        ":return: The encoded message and timestamp.\n");
+
+PyDoc_STRVAR(
+        encode_message_doc,
+        "encode_message(msg)\n"
+        "--\n\n"
+        "Encodes the log `msg`.\n"
+        ":param msg: Log message to encode.\n"
+        ":raises NotImplementedError: If the log message failed to encode.\n"
+        ":return: The encoded message.\n");
+
+PyDoc_STRVAR(
+        encode_timestamp_delta_doc,
+        "encode_timestamp_delta(timestamp_delta)\n"
+        "--\n\n"
+        "Encodes the timestamp.\n"
+        ":param timestamp_delta: Timestamp difference in miliseconds between the current log "
+        "message and the previous log message.\n"
+        ":raises NotImplementedError: If the timestamp failed to encode.\n"
+        ":return: The encoded timestamp.\n");
+
+PyDoc_STRVAR(module_doc, "Python interface to the CLP IR four byte encoding methods.");
+
+/**
+ * Method table
+ */
+static PyMethodDef encoder_method[]{
+        {"encode_preamble",
+         clp_ffi_py::encoder::four_byte_encoding::encode_preamble,
+         METH_VARARGS,
+         static_cast<char const*>(encode_preamble_doc)},
+
+        {"encode_message_and_timestamp_delta",
+         clp_ffi_py::encoder::four_byte_encoding::encode_message_and_timestamp_delta,
+         METH_VARARGS,
+         static_cast<char const*>(encode_message_and_timestamp_delta_doc)},
+
+        {"encode_message",
+         clp_ffi_py::encoder::four_byte_encoding::encode_message,
+         METH_VARARGS,
+         static_cast<char const*>(encode_message_doc)},
+
+        {"encode_timestamp_delta",
+         clp_ffi_py::encoder::four_byte_encoding::encode_timestamp_delta,
+         METH_VARARGS,
+         static_cast<char const*>(encode_timestamp_delta_doc)},
+
+        {nullptr, nullptr, 0, nullptr}};
+
+/**
+ * Module definition
+ */
+static PyModuleDef clp_four_byte_encoder{
+        PyModuleDef_HEAD_INIT,
+        "CLPFourByteEncoder",
+        static_cast<char const*>(module_doc),
```

Comment:
Am I missing something otherwise these `static_cast`s are unnecessary - `module_doc` (and other similarly defined variables) are `char const` arrays which will decay to the corresponding pointer types. Similarly for `PyMethodDef encoder_method` (you don't need to `static_cast`) - IMO, seeing these casts make me think something else is going on

Reply:
An error is sent by clang-tidy if I'm not doing the casting: "warning: do not implicitly decay an array into a pointer; consider using gsl::array_view or an explicit cast instead"
Even though I turned off the clang-tidy for this file, but before doing that, I tried to fix all the reasonable and easy-to-fix warnings. This is one of them.